src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -35,4 +45,21 @@ public SchemaService(final SchemaRepository schemaRepository,
                         () -> schemaRepository.getSchemasCount(name)));
     }
 
+    public Result<EventTypeSchema> getSchemaVersion(final String name, final String version) {
+        final Matcher versionMatcher = VERSION_PATTERN.matcher(version);
```

Comment:
@adyach since we are using Spring, I would just follow the Spring way of validating format. It supports these annotations which goes into the controller. Service layer is for business rules and data format is understood by spring to be a user interfacing responsibility, e.g. controller stuff.